### PR TITLE
refactor(in-4193): use async hooks in multistore extension

### PR DESCRIPTION
**Pull request by lsliwaradioluz**
### 🔗 Linked issue

[IN-4193](https://alokai.atlassian.net/browse/IN-4193)

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSDoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

[ADDED] Support for asynchronous `fetchConfiguration()` and cacheManagerFactory's `get()` / `set()` methods.

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/multistore/src/types.ts [hunk 1]
```diff
@@ -3,19 +3,24 @@
  */
 export type StoreConfig = Record<string, any>;
 
+/**
+ * Utility for typing functions which can be defined as either async or sync.
+ */
+type MaybePromise<T> = T | Promise<T>;
```
Comment (lukasborawski) [inline packages/multistore/src/types.ts:9]: Isn't it better to use `Awaited`?

**Review by lukasborawski — APPROVED** [review]

File: packages/multistore/src/types.ts (hunk 1 above)
Comment (lsliwaradioluz) [reply packages/multistore/src/types.ts:9]: So far I've been using `Awaited` only to extract the return type of async functions, not to type them. I am not sure how does `Awaited` fit in here. Could you show a full example?

File: packages/multistore/src/extension.ts [hunk 2]
```diff
@@ -1,3 +1,5 @@
+// eslint-disable-next-line import/no-extraneous-dependencies
```
[the remark this answers is not in the document]
Comment (lsliwaradioluz) [reply packages/multistore/src/extension.ts]: [Added](https://github.com/vuestorefront/vue-storefront/pull/7183/commits/a15c857e2d582383b1a6df651117fe3fe18dba3d)

[the remark this answers is not in the document]
Comment (lsliwaradioluz) [reply packages/multistore/src/extension.ts]: you are right, [done](https://github.com/vuestorefront/vue-storefront/pull/7183/commits/eb5893e508918aee51ad54aa095d7d9d5ba075b1)

**Pull request merged**